boot/zephyr/main.c

```diff
@@ -175,6 +183,40 @@ static void do_boot(struct boot_rsp *rsp)
 }
 #endif
 
+#if defined(CONFIG_LOG) && !defined(CONFIG_LOG_IMMEDIATE) &&\
+    !defined(CONFIG_LOG_PROCESS_THREAD)
+#include <logging/log_ctrl.h>
+
+/* most simple log processing theread */
+void boot_log_therad_func(void *dummy1, void *dummy2, void *dummy3)
+{
+    (void)dummy1;
+    (void)dummy2;
+    (void)dummy3;
+
+     log_init();
+
+     while (1) {
+             if (log_process(false) == false) {
+                    k_sleep(30);
+             }
+     }
+}
+
+K_THREAD_STACK_DEFINE(boot_log_stack, 768);
+struct k_thread boot_log_thread;
+
+void zephyr_boot_log_start(void)
+{
+		/* start logging thread */
+		k_thread_create(&boot_log_thread, boot_log_stack,
+				K_THREAD_STACK_SIZEOF(boot_log_stack),
+				boot_log_therad_func, NULL, NULL, NULL,
+				K_HIGHEST_APPLICATION_THREAD_PRIO, 0, 30);
+		k_thread_name_set(&boot_log_thread, "logging");
+}
+#endif
```

Comment:
```suggestion
#endif /* defined(CONFIG_LOG) && !defined(CONFIG_LOG_IMMEDIATE) &&\
    !defined(CONFIG_LOG_PROCESS_THREAD) */
```

Reply:
Not sure if this is really needed, but some comment on what the condition for the corresponding `if` would be welcome IMO